test/parallel/test-assert-esm-cjs-message-verify.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+require('../common');
+const tmpdir = require('../common/tmpdir');
+const assert = require('assert');
+const { writeFileSync, unlink } = require('fs');
+const { execSync } = require('child_process');
+
+tmpdir.refresh();
+
+// To ensure the assert ok throwing similar error messages for esm and cjs
+
+const nodejs = `${process.execPath}`;
+
+// Importing package for each extensions.
+const fileImports = {
+  mjs: 'import assert from "assert";',
+  js: 'const assert = require("assert");',
+};
+
+const fileNames = [];
+
+for (const [ext, header] of Object.entries(fileImports)) {
+  const fileName = `test-file.${ext}`;
+  // Store the generated filesnames in an array
+  fileNames.push(`${tmpdir.path}/${fileName}`);
+
+  writeFileSync(tmpdir.resolve(fileName), `${header}\nassert.ok(0 === 2);`);
+}
+
+const errorsMessages = [];
+
+for (const fileName of fileNames) {
+  const command = `${nodejs} ${fileName}`;
+  assert.throws(
+    () => {
+      execSync(command, { stdio: 'pipe' });
+    },
+    (e) => {
+      // For each error message, filter the lines which will starts with AssertionError
+      errorsMessages.push(
+        e.message.split('\n').find((s) => s.startsWith('AssertionError'))
+      );
+      return true;
+    }
+  );
```

Comment:
As you've written the test, you assume there won't be any shell special char to escape in the path. We can avoid this pitfall by using spawn – e.g. with `common.spawnPromisified`:
```suggestion
  const { stderr, code } = await spawnPromisified(process.execPath, [fileName]);
  assert.strictEqual(code, 1);
  // For each error message, filter the lines which will starts with AssertionError
  errorsMessages.push(
        stderr.split('\n').find((s) => s.startsWith('AssertionError'))
  );
```

Reply:
Its a race condition. But, still its good suggestion to use the `spawnPromisified` over the `exec`. I will include them on rewrite of the test file.